legend-engine-test-runner-mapping/src/main/java/org/finos/legend/engine/testable/mapping/extension/MappingTestRunner.java

```diff
@@ -92,121 +96,88 @@ public TestResult executeAtomicTest(Root_meta_pure_test_AtomicTest atomicTest, P
         try
         {
-            Optional<InputStream> inputStream = buildInputStreamIfPossible(dataTest, context);
-            if (!inputStream.isPresent())
+            Assert.assertTrue(mappingTest.assertions.size() == 1, () -> "Mapping Test expected to have one assertion");
+            Root_meta_pure_runtime_Runtime_Impl runtime = new Root_meta_pure_runtime_Runtime_Impl("");
+            List<Pair<String, EmbeddedData>> connectionInfo = mappingTest.storeTestData.stream().map(testData -> Tuples.pair(testData.store, EmbeddedDataHelper.resolveEmbeddedDataInPMCD(context.getPureModelContextData(), testData.data))).collect(Collectors.toList());
+            connections = connectionInfo.stream()
+                    .map(pair -> this.factories.collect(f -> f.tryBuildTestConnectionsForStoreWithMultiInputs(context.getDataElementIndex(), resolveStore(context.getPureModelContextData(), pair.getOne()), pair.getTwo())).select(Objects::nonNull).select(Optional::isPresent)
+                            .collect(Optional::get).getFirstOptional().orElseThrow(() -> new UnsupportedOperationException("Unsupported store type for:'" + pair.getOne() + "' mentioned while running the mapping tests"))).collect(Collectors.toList());
+            connections.forEach(conn -> runtime._connectionsAdd(conn.getOne().accept(context.getConnectionVisitor())));
+
+            Optional<String> executeInput = resolveInputStringFromDataElement(mappingTest, context);
+            SingleExecutionPlan executionPlan = context.getPlan();
+            if (executionPlan == null || !executeInput.isPresent())
             {
-                throw new UnsupportedOperationException("Unable to build input stream with data from mapping data test " + "'" + dataTest.id + "'");
+                executionPlan = PlanGenerator.generateExecutionPlan(context.getMetamodelTestSuite()._query(), pureMapping, runtime, null, context.getPureModel(), this.pureVersion, PlanPlatform.JAVA, null, context.getRouterExtensions(), context.getExecutionPlanTransformers());
+                context.setPlan(executionPlan);
             }
-            inputStream.ifPresent(stream ->  context.getExecuteBuilder().withInputAsStream(stream));
-            return executeTestAssertions(dataTest,context.getExecuteBuilder());
+            executeInput.ifPresent(stream ->  context.getExecuteBuilder().withInputAsString(stream));
+
+            // execute assertion
+            TestAssertion assertion = mappingTest.assertions.get(0);
+            PlanExecutor.ExecuteArgs executeArgs = context.getExecuteBuilder().build();
+            Result result = this.executor.executeWithArgs(executeArgs);
+            if (result instanceof StreamingResult)
+            {
+                // We want to read streaming results only once
+                StreamingResult streamingResult = (StreamingResult) result;
+                result = new ConstantResult((streamingResult.flush(streamingResult.getSerializer(SerializationFormat.RAW))));
```

Comment:
Test Assertion Evaluator already does this, no?

Reply:
i think i can remove this since its only one assertion. but if more assertions we need to have this because we can only read the result once